tests/cache/models/aspects/deleteTags.js

```diff
@@ -0,0 +1,141 @@
+const oneDeletePath = '/v1/aspects/{key}/tags/{akey}';
+const redisOps = require('../../../../cache/redisOps');
+const objectType = require('../../../../cache/sampleStore')
+                    .constants.objectType;
+const samstoinit = rtu.samstoinit;
+
+describe(`api: redisStore: aspects: DELETE tags`, () => {
+  let token;
+  let aspId;
+  let aspName;
+  const tag0 = 'tag0';
+
+  const n = {
+    name: `${tu.namePrefix}ASPECTNAME`,
+    timeout: '110s',
+    tags: ['tag0', 'tag1'],
+    isPublished: true,
+  };
+
+  before((done) => {
+    tu.toggleOverride('enableRedisSampleStore', true);
+    tu.createToken()
+    .then((returnedToken) => {
+      token = returnedToken;
+      done();
+    })
+    .catch(done);
+  });
+
+  beforeEach((done) => {
+    Aspect.create(n)
+    .then((asp) => {
+      aspId = asp.id;
+      aspName = asp.name;
+      return samstoinit.eradicate();
+    })
+    .then(() => samstoinit.init())
+    .then(() => done())
+    .catch(done);
+  });
+  afterEach(rtu.forceDelete);
+  after(() => tu.toggleOverride('enableRedisSampleStore', false));
+
+  it('delete all tags', (done) => {
+    api.delete(allDeletePath.replace('{key}', aspId))
+    .set('Authorization', token)
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        done(err);
+      }
+      expect(res.body.tags).to.have.length(0);
+      redisOps.getHashPromise(objectType.aspect, n.name)
+      .then((aspect) => {
+        expect(JSON.parse(aspect.tags)).to.have.length(0);
+        done();
+      });
+    });
+  });
+
+  it('delete one tag', (done) => {
+    api.delete(oneDeletePath.replace('{key}', aspId).replace('{akey}', tag0))
+    .set('Authorization', token)
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        done(err);
+      }
+      expect(res.body.tags).to.have.length(1);
+      expect(res.body.tags).to.have.members(['tag1']);
+      redisOps.getHashPromise(objectType.aspect, n.name)
+      .then((aspect) => {
+        const tags = JSON.parse(aspect.tags);
+        expect(tags).to.have.length(1);
+        expect(tags).to.have.members(['tag1']);
+        done();
+      });
+    });
+  });
+
+  it('delete tag by name', (done) => {
+    api.delete(oneDeletePath.replace('{key}', aspName).replace('{akey}', tag0))
+    .set('Authorization', token)
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        done(err);
+      }
+      expect(res.body.tags).to.have.length(1);
+      expect(res.body.tags).to.have.members(['tag1']);
+      redisOps.getHashPromise(objectType.aspect, n.name)
+      .then((aspect) => {
+        const tags = JSON.parse(aspect.tags);
+        expect(tags).to.have.length(1);
+        expect(tags).to.have.members(['tag1']);
+        done();
+      });
+    });
+  });
+
+  it('error if tag not found', (done) => {
+    api.delete(oneDeletePath.replace('{key}', aspId).replace('{akey}', 'x'))
```

Comment:
I just copied this test from api and checked the redis entry. I dont think we throw an error if tag does not exist. Is that correct @shriramshankar ? If that is true, I can change the description of test.